sql/analyzer/indexed_joins.go

```diff
@@ -369,18 +375,31 @@ func convertSemiToInnerJoin(a *Analyzer, m *memo.Memo) error {
 
 		// project belongs to the original group
 		leftCols := semi.Left.RelProps.OutputCols()
-		var projections []*memo.ExprGroup
-		for i := range leftCols {
-			col := leftCols[i]
-			if col.Name == "" && col.Source == "" {
-				continue
+		var projections []sql.Expression
+		for colId, hasNext := leftCols.Next(1); hasNext; colId, hasNext = leftCols.Next(colId + 1) {
```

Comment:
This is an odd interface. `Next` doesn't usually take any input.

Either use a) `Iter(callback func()` or b) `cols.Iter()` which returns an iterator object with `Next()` and `HasNext()` methods

Reply:
The bitset's iterator value also being its iter position seems particularly elegant and simpler than creating intermediate objects, it's like indexing into a list but better. We would need multiple new interfaces to cover all of the ways this is used. I think we should preserve the way bitsets are simpler than arrays.